docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,222 @@
+
+Nevertheless, `odo` will provide a way to manipulate this file via a CLI command, as described in [managing configuration](#managing-configuration)
+
+### <a name="managing-configuration"></a> Managing configuration
+
+Developers will need to add configuration values to the `.odo` config files. This configuration management will be managed by a new command in `odo` CLI. As configuration will be of different types, there will be needs to additional verbs (to the regular ones) for managing this configuration.
+
+#### Creating configuration
+
+There might be times that a developer would want to add/edit/remove values from the .odo configuration file. There should be a command to manage this file in the `odo` CLI. That command should be the same used to touch configuration of the global `odo` config file located in the users's home directory.
+
+Currently that command is:
+
+```bash
+odo utils config [create|update|delete|list|add|remove|get|describe] [KEY] [VALUES] [--local|--local]
+```
+
+In order to create configuration, we should call odo with the appropriate key and value or list of values to apply, as well as the scope for this configuration to be saved (local or global).
+
+```bash
+odo utils config set memory 2Gi               # Set's component memory to 2 GB local config
+odo utils config add odoignore .odo --local   # Adds a value to odoignore list local config
+odo utils config add odoignore .odo --global  # Adds a value to odoignore list global config
+```
+
+As can be seen, when a scope is ommited, the value will be applied to the local configuration store.
+
+The verbs that add configuration are:
+
+* *create*: Creates a configuration entry with the value provided. If the key exists, this command should error with the corresponding message.
+* *add*: Adds a value to an existing config entry. The type of the entry should be a list. If the key does not exist, or is not a list, this command should error with the corresponding error message.
+
+*NOTE*: When creating a component, no configuration file will be created. If a developer thinks a configuration value is worth being stored in the `.odo` configuration file, he will need to use the previous command to create these files (whether global or local doesn't matter).
+
+#### Listing configuration
+
+When a user wants to know what specific configuration exists, all or just a specific key.
+
+```bash
+odo utils config list --global      # List all global config
+odo utils config get memory         # List the value for memory config key
+```
+
+The verbs that add configuration are:
+
+* *list*: Lists all configuration. No further argument.
+* *get*: Lists configuration for a specific entry. If the key does not exist this command should error with the corresponding message.
+
+#### Updating configuration
+
+When a user wants to update specific configuration.
+
+```bash
+odo utils config list --global                # List all global config
```

Comment:
should this be odo update example here instead?

Reply:
why `odo update`?

Reply:
sorry I meant `odo utils config update ...` instead of `odo utils config list` in the example